compilation error when stdnoreturn.h is included

Hi,

when `stdnoreturn.h` is included before `unity.h` a compiler error for a unknown attribute is generated:
```
In file included from ../../src/unity.h:21,
                 from ../tests/test_unity_strings.c:7:
../../src/unity_internals.h:621:1: error: '_Noreturn' attribute directive ignored [-Werror=attributes]
  621 | void UnityFail(const char* message, const UNITY_LINE_TYPE line) UNITY_FUNCTION_ATTR(noreturn);
      | ^~~~
../../src/unity_internals.h:622:1: error: '_Noreturn' attribute directive ignored [-Werror=attributes]
  622 | void UnityIgnore(const char* message, const UNITY_LINE_TYPE line) UNITY_FUNCTION_ATTR(noreturn);
      | ^~~~
```

This seems to occur because in `stdnoreturn.h` `noreturn` is defined to the keyword `_Noreturn`.

--- a/src/unity_internals.h
+++ b/src/unity_internals.h
@@ -46,6 +46,20 @@
   #define UNITY_FUNCTION_ATTR(a)    /* ignore */
 #endif
 
+#ifndef UNITY_NORETURN
+  #if defined(__cplusplus)
+    #if __cplusplus >= 201103L
+      #define UNITY_NORETURN [[ noreturn ]]
+    #endif
+  #elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+    #include <stdnoreturn.h>
+    #define UNITY_NORETURN noreturn
+  #endif
+#endif
+#ifndef UNITY_NORETURN
+  #define UNITY_NORETURN UNITY_FUNCTION_ATTR(noreturn)
+#endif
+
 /*-------------------------------------------------------
  * Guess Widths If Not Specified
  *-------------------------------------------------------*/
@@ -618,8 +632,8 @@ void UnityAssertNumbersArrayWithin(const UNITY_UINT delta,
                                    const UNITY_FLAGS_T flags);
 
 #ifndef UNITY_EXCLUDE_SETJMP_H
-void UnityFail(const char* message, const UNITY_LINE_TYPE line) UNITY_FUNCTION_ATTR(noreturn);
-void UnityIgnore(const char* message, const UNITY_LINE_TYPE line) UNITY_FUNCTION_ATTR(noreturn);
+UNITY_NORETURN void UnityFail(const char* message, const UNITY_LINE_TYPE line);
+UNITY_NORETURN void UnityIgnore(const char* message, const UNITY_LINE_TYPE line);
 #else
 void UnityFail(const char* message, const UNITY_LINE_TYPE line);
 void UnityIgnore(const char* message, const UNITY_LINE_TYPE line);

